components/service/glean/src/main/java/mozilla/components/service/glean/ping/MetricsPing.kt

```diff
@@ -0,0 +1,40 @@
+package mozilla.components.service.glean.ping
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.support.annotation.VisibleForTesting
+import java.util.concurrent.TimeUnit
+
+class MetricsPing(var applicationContext: Context) {
+    @VisibleForTesting
+    internal val metricsPreferences: SharedPreferences by lazy { getMetricsPrefs() }
+
+    companion object {
+        const val LAST_METRICS_PING_TIMESTAMP_KEY = "last_metrics_ping_timestamp"
+        const val STORE_NAME = "metrics"
+        const val DOC_TYPE = "metrics"
+    }
+
+    @Suppress("MagicNumber")
+    fun has24HoursPassed(): Boolean {
+        val lastPingTime = metricsPreferences.getLong(LAST_METRICS_PING_TIMESTAMP_KEY, 0L)
```

Comment:
Let's also add a test to make sure that we don't break if `LAST_METRICS_PING_TIMESTAMP_KEY` contains garbage (e.g. a string)

Reply:
I've changed this to be protected with a try/catch which will catch a ClassCastException which is what getLong throws if the value is not a Long.